test/integration/DCAHubCompanion/multi-call.spec.ts

```diff
@@ -87,15 +85,210 @@ contract('Multicall', () => {
+            await DCAHubCompanion.multicall([permissionData, reduceData, swapExecutionData, sendData]);
+            minExpected = expectedAmountOut;
+          });
+          if (to === 'USDC') {
+            then('recipient has expected amount of USDC balance', async () => {
+              const balance = await USDC.balanceOf(recipient.address);
+              expect(balance.gte(minExpected)).to.be.true;
+            });
+          } else {
+            then('recipient has expected amount of ETH balance', async () => {
+              const balance = await ethers.provider.getBalance(recipient.address);
+              expect(balance.gte(recipientInitialETHBalance.add(minExpected))).to.be.true;
+            });
+          }
+          thenCompanionRemainsWithoutAnyBalance();
+        });
+      }
+    });
+
+    describe('terminate and swap', () => {
+      when('terminating a position and swapping WETH to ETH, and USDC to wBTC', () => {
+        let expectedAmountOutETH: BigNumber, expectedAmountOutBTC: BigNumber;
+        given(async () => {
+          const { positionId, swappedBalance, unswappedBalance } = await depositAndSwap({ from: WETH, to: USDC, amount: ETH_1 });
+          const permissionData = await givePermissionToCompanionData({ signer: positionOwner, positionId, permissions: [Permission.TERMINATE] });
+          const terminateData = await terminateAndSendToCompanionData({ positionId });
+          const { swapExecutionData: executionDataETH, expectedAmountOut: _expectedAmountOutETH } = await runSwapData({
+            tokenIn: WETH.address,
+            amountIn: unswappedBalance,
+            swap: ({ amountIn }) => transformWETHToETH(amountIn),
+          });
+          const { swapExecutionData: executionDataWBTC, expectedAmountOut: _expectedAmountOutBTC } = await runSwapData({
+            tokenIn: USDC.address,
+            amountIn: swappedBalance,
+            swap: ({ amountIn }) => swapIn0x({ from: USDC, to: WBTC, amountIn }),
+          });
+          const sendETHData = await sendAllInCompanionToRecipientData({ token: await DCAHubCompanion.PROTOCOL_TOKEN(), recipient });
+          const sendWBTCData = await sendAllERC20InCompanionToRecipientData({ token: WBTC, recipient });
+          await DCAHubCompanion.multicall([permissionData, terminateData, executionDataETH, sendETHData, executionDataWBTC, sendWBTCData]);
+          expectedAmountOutETH = _expectedAmountOutETH;
+          expectedAmountOutBTC = _expectedAmountOutBTC;
+        });
+        then('recipient has expected amount of wBTC balance', async () => {
+          const balance = await WBTC.balanceOf(recipient.address);
+          expect(balance.gte(expectedAmountOutBTC)).to.be.true;
+        });
+        then('recipient has expected amount of ETH balance', async () => {
+          const balance = await ethers.provider.getBalance(recipient.address);
+          expect(balance.gte(recipientInitialETHBalance.add(expectedAmountOutETH))).to.be.true;
+        });
```

Comment:
Not checking position was terminated, but we can induce it from all the other info tho so I'm cool with that.